docs/src/main/asciidoc/extension-authors-guide.adoc

```diff
@@ -340,6 +342,52 @@ You will also need to configure the `maven-compiler-plugin` to detect the `quark
 </build>
 ----
 
+==== Create new extension modules using Maven
+
+`create-extension` mojo of `io.quarkus:quarkus-maven-plugin` can be used to generate stubs of Maven modules needed for implementing a new Quarkus extension.
+
+WARNING: This mojo can be currently used only for adding extensions to an established source tree, such as https://github.com/quarkusio/quarkus[Quarkus] or https://github.com/quarkusio/quarkus[Camel Quarkus]. Creating extension projects from scratch is not supported yet.
+
+Example: Let's add a new extension called `my-ext` to Quarkus source tree:
+
+[source, shell, subs=attributes+]
+----
+git clone https://github.com/quarkusio/quarkus.git
+cd quarkus
+cd extensions
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create-extension -N -Dquarkus.artifactIdBase=my-ext
+----
+
+The above sequence of commands will create three new Maven modules:
+
+* `quarkus-my-ext-parent` in `my-ext` directory; this module is also added to `<modules>` of `quarkus-extensions-parent`
+* `quarkus-my-ext` in `my-ext/runtime` directory
+* `quarkus-my-ext-deployment` in `my-ext/deployment` directory; a basic `MyExtProcessor` class is generated in this module.
+
+Trying to build the new extension without any change will result in Maven complaining about the missing version of `quarkus-my-ext` in `quarkus-my-ext-deployment`. So solve that, we need to add `quarkus-my-ext` to the `dependencyManagement` in `bom/runtime/pom.xml`. After that, the build of the new extension modules should pass flawlessly.
+
+Note that those parameters of the mojo that will be constant for all extensions added in this source tree are configured in `extensions/pom.xml` so that they do not need to be passed on the command line each time a new extension is added.
```

Comment:
```suggestion
Note that the parameters of the mojo that will be constant for all the extensions added to this source tree are configured in `extensions/pom.xml` so that they do not need to be passed on the command line each time a new extension is added.
```

Reply:
Fixed in b4067d0